### PR TITLE
Allow users to configure name assigned to CLI configs in auto-setup

### LINKED ISSUE #474: Error: flag name is required if you already have existing configs
Hi,
I'm running a docker-compose with grafana on a new machine.
My only env variables are the following (which normally are sufficient to work)

```
environment:
      - DOCKER_INFLUXDB_INIT_MODE=setup
      - DOCKER_INFLUXDB_INIT_USERNAME=${INFLUXDB_USERNAME}
      - DOCKER_INFLUXDB_INIT_PASSWORD=${INFLUXDB_PASSWORD}
      - DOCKER_INFLUXDB_INIT_ORG=${INFLUXDB_ORG}
      - DOCKER_INFLUXDB_INIT_BUCKET=${INFLUXDB_BUCKET}
      - DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=${INFLUXDB_TOKEN}
      - INFLUXDB_REPORTING_DISABLED=true
```

but i receive the error:

`Error: flag name is required if you already have existing configs`

Below the full log

```
influxdb2   | 2021-04-01T07:41:27.387733750Z    info    pinging influxd...      {"system": "docker"}
influxdb2   | ts=2021-04-01T07:41:28.377067Z lvl=info msg="Welcome to InfluxDB" log_id=0TF5vej0000 version=2.0.4 commit=4e7a59bb9a build_date=2021-02-08T17:47:02Z
influxdb2   | ts=2021-04-01T07:41:28.384552Z lvl=info msg="Resources opened" log_id=0TF5vej0000 service=bolt path=/var/lib/influxdb2/influxd.bolt
influxdb2   | ts=2021-04-01T07:41:28.386351Z lvl=info msg="Bringing up metadata migrations" log_id=0TF5vej0000 service=migrations migration_count=14
influxdb2   | ts=2021-04-01T07:41:28.543369Z lvl=info msg="Using data dir" log_id=0TF5vej0000 service=storage-engine service=store path=/var/lib/influxdb2/engine/data
influxdb2   | ts=2021-04-01T07:41:28.543547Z lvl=info msg="Compaction settings" log_id=0TF5vej0000 service=storage-engine service=store max_concurrent_compactions=4 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
influxdb2   | ts=2021-04-01T07:41:28.543578Z lvl=info msg="Open store (start)" log_id=0TF5vej0000 service=storage-engine service=store op_name=tsdb_open op_event=start
influxdb2   | ts=2021-04-01T07:41:28.543666Z lvl=info msg="Open store (end)" log_id=0TF5vej0000 service=storage-engine service=store op_name=tsdb_open op_event=end op_elapsed=0.090ms
influxdb2   | ts=2021-04-01T07:41:28.543740Z lvl=info msg="Starting retention policy enforcement service" log_id=0TF5vej0000 service=retention check_interval=30m
influxdb2   | ts=2021-04-01T07:41:28.543806Z lvl=info msg="Starting precreation service" log_id=0TF5vej0000 service=shard-precreation check_interval=10m advance_period=30m
influxdb2   | ts=2021-04-01T07:41:28.543915Z lvl=info msg="Starting query controller" log_id=0TF5vej0000 service=storage-reads concurrency_quota=10 initial_memory_bytes_quota_per_query=9223372036854775807 memory_bytes_quota_per_query=9223372036854775807 max_memory_bytes=0 queue_size=10
influxdb2   | ts=2021-04-01T07:41:28.545182Z lvl=info msg="Configuring InfluxQL statement executor (zeros indicate unlimited)." log_id=0TF5vej0000 max_select_point=0 max_select_series=0 max_select_buckets=0
influxdb2   | ts=2021-04-01T07:41:28.846783Z lvl=info msg=Starting log_id=0TF5vej0000 service=telemetry interval=8h
influxdb2   | ts=2021-04-01T07:41:28.846848Z lvl=info msg=Listening log_id=0TF5vej0000 transport=http addr=:9999 port=9999
influxdb2   | 2021-04-01T07:41:29.405788062Z    info    pinging influxd...      {"system": "docker"}
influxdb2   | 2021-04-01T07:41:29.446345062Z    info    got response from influxd, proceeding   {"system": "docker"}
influxdb2   | Error: flag name is required if you already have existing configs
influxdb2   | See 'influx setup -h' for help
influxdb2   | 2021-04-01T07:41:29.476713112Z    warn    cleaning bolt and engine files to prevent conflicts on retry    {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt", "engine_path": "/var/lib/influxdb2/engine"}
influxdb2 exited with code 1
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #474

### CODE REVIEW

File: influxdb/test/cases/test-auto-setup-custom-config-name [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    -e DOCKER_INFLUXDB_INIT_CLI_CONFIG_NAME=test
+    influxdb:${tag} influxd
+)
+
+log_msg Booting 2.x container in setup mode
+if ! ${docker_run_influxd[@]} > /dev/null; then
+    log_msg Error: Failed to launch container
+    exit 1
+fi
+wait_container_ready
```
Comment (wbaker85) [inline influxdb/test/cases/test-auto-setup-custom-config-name:65]: Is there any way you can verify that the CLI config name created matches the one supplied on line 56 with this test?

File: influxdb/2.0/Dockerfile [hunk 2]
```diff
@@ -60,3 +60,4 @@ EXPOSE 8086
 ENV INFLUX_CONFIGS_PATH /etc/influxdb2/influx-configs
 ENV INFLUXD_INIT_PORT 9999
 ENV INFLUXD_INIT_PING_ATTEMPTS 600
+ENV DOCKER_INFLUXDB_INIT_CLI_CONFIG_NAME default
```
Comment (wbaker85) [inline influxdb/2.0/Dockerfile:63]: Why does this need to be set here for the `CLI_CONFIG_NAME`, whereas other `DOCKER_INFLUXDB_INIT_...` variables aren't?

Comment (danxmoran) [reply influxdb/2.0/Dockerfile:63]: None of the other `DOCKER_INFLUXDB_INIT_` variables have defaults. It is odd to see all the different prefixes here, but I like having all the defaults in one place

File: influxdb/test/cases/test-auto-setup-custom-config-name (hunk 1 above)
Comment (danxmoran) [reply influxdb/test/cases/test-auto-setup-custom-config-name:65]: 👍 I can do something simple like grep for it

Comment (danxmoran) [reply influxdb/test/cases/test-auto-setup-custom-config-name:70]: Added

**Review by wbaker85 — APPROVED** [review]

**Pull request merged**